src/mscorlib/corefx/System/Buffers/ArrayPool.cs

```diff
@@ -19,6 +19,16 @@ namespace System.Buffers
     /// </remarks>
     public abstract class ArrayPool<T>
     {
+        /// <summary>Per-type cached pool.</summary>
+        /// <remarks>
+        /// byte[] and char[] are the most commonly pooled array types. For these we use a special pool type
+        /// optimized for very fast access speeds, at the expense of more memory consumption.
+        /// The shared instance is created lazily on first access by the runtime.
+        /// </remarks>
+        private readonly static ArrayPool<T> _sharedInstance =
+            typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
+            Create();
```

Comment:
Why did this get moved?  This means that if someone does:
```C#
ArrayPool<MyCustomType> pool = ArrayPool.Create<MyCustomType>();
```
and never uses ```ArrayPool<MyCustomType>.Shared```, we're still going to create an instance of ```ArrayPool<MyCustomType>``` in Create to be stored in ```_sharedInstance```.